src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java

```diff
@@ -170,7 +173,14 @@ private void invokeFileNio(DataOperation<? extends DataItem> fileOperation) {
 					//TODO: EventStreamWriter<ByteBuffer>  ->  writeEvent
 					break;
 				case READ:
-					//TODO: EventStreamReader<ByteBuffer>  ->  readNextEvent
+					if ((null != scopeMap.get(scopeName) && null != scopeMap.get(scopeName).get(streamName))) {
+						if (invokeDataRead(fileOperation, scopeName, streamName)) {
```

Comment:
looks like something should be done if condition is `true`

Reply:
nothing should be done, DataRead is caused in condition.
we can make some logs if returns TRUE

Reply:
we agreed that we don't this check. Do we need it now?

Reply:
If first condition is FALSE, the second will be never executed.
So, there will be no exception. (as I understand)
"short-circuiting" behavior in Java

Reply:
We don't need the check with maps at all.